### PR TITLE
CI: check docker requires

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

adding available requirements check to dockers

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by akihironitta — APPROVED** [review]

File: .github/workflows/ci_dockers.yml [hunk 1]
```diff
@@ -73,9 +73,10 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        # the config used in '.azure-pipelines/gpu-tests.yml'
-        python_version: ["3.7"]
-        pytorch_version: ["1.8"]
+        include:
+          # the config used in '.azure-pipelines/gpu-tests.yml'
+          - {python_version: "3.7", pytorch_version: "1.8"}
+          - {python_version: "3.9", pytorch_version: "1.10"}
```
Comment (akihironitta) [inline .github/workflows/ci_dockers.yml:79]: Two questions:
- Where will (py3.9, pt1.10) be used?
- Do you think it's reasonable to test (py3.9, pt1.11) instead of (py3.9, pt1.10) here?

```suggestion
          - {python_version: "3.9", pytorch_version: "1.11"}
```

Comment (Borda) [reply .github/workflows/ci_dockers.yml:79]: it is used in Tutorials:
https://github.com/PyTorchLightning/lightning-tutorials/blob/0bb4a7e4ba5053a36c9bf229447f5332c9d848f4/.azure/ipynb-publish.yml#L24
and I may say that 1.10 is more stable then the recent 1.11 (lets update later)

**Review by awaelchli — APPROVED** [review]

**Review by krshrimali — APPROVED** [review]
🚀

File: .azure-pipelines/gpu-tests.yml [hunk 2]
```diff
@@ -52,9 +52,10 @@ jobs:
 
     - bash: |
         python -c "fname = 'requirements/extra.txt' ; lines = [line for line in open(fname).readlines() if 'horovod' not in line] ; open(fname, 'w').writelines(lines)"
-        pip install fairscale==0.4.5
+        pip install fairscale>=0.4.5
         pip install deepspeed>=0.6.0
-        pip install bagua-cuda102==0.9.0
+        CUDA_VERSION_MM=$(python -c "import torch ; print(''.join(map(str, torch.version.cuda.split('.')[:2])))")
```
Comment (carmocca) [inline .azure-pipelines/gpu-tests.yml:57]: Isn't this simpler?

```suggestion
        CUDA_VERSION_MM=$(python -c "import torch ; print(torch.version.cuda.replace('.', ''))")
```

File: dockers/base-conda/Dockerfile [hunk 3]
```diff
@@ -147,12 +147,33 @@ RUN \
...
+    python -c "import fairscale; print(fairscale.__version__)"
+
+RUN \
+    # install DeepSpeed
+    pip install deepspeed==0.6.0 && \
+    python -c "import deepspeed; print(deepspeed.__version__)"
+
+RUN \
+    # install Bagua
+    CUDA_VERSION_MM=$(python -c "print(''.join('$CUDA_VERSION'.split('.')[:2]))") && \
```
Comment (carmocca) [inline dockers/base-conda/Dockerfile:162]: same comment

File: dockers/base-conda/Dockerfile [hunk 4]
```diff
@@ -147,12 +147,33 @@ RUN \
...
+    pip install "bagua-cuda$CUDA_VERSION_MM==0.9.0" && \
+    python -c "import bagua_core; bagua_core.install_deps()" && \
```
Comment (carmocca) [inline dockers/base-conda/Dockerfile:164]: I recently installed bagua and didn't have to do this :thinking:

File: dockers/base-conda/Dockerfile [hunk 5]
```diff
@@ -147,12 +147,33 @@ RUN \
     pip install --no-cache-dir --global-option="--cuda_ext" https://github.com/NVIDIA/apex/archive/refs/heads/master.zip && \
     python -c "from apex import amp"
 
+RUN \
+    # install FairScale
+    pip install fairscale==0.4.5 && \
```
Comment (carmocca) [inline dockers/base-conda/Dockerfile:152]: In the gpu job you set it to `>=`. Shouldn't they match?

File: dockers/base-cuda/Dockerfile [hunk 6]
```diff
@@ -76,7 +76,6 @@ RUN \
     pip install -q fire && \
     # Disable cache \
     CUDA_VERSION_MM=$(python -c "print(''.join('$CUDA_VERSION'.split('.')[:2]))") && \
```
Comment (carmocca) [inline dockers/base-cuda/Dockerfile:78]: Remove this too?

File: dockers/base-cuda/Dockerfile [hunk 7]
```diff
@@ -138,14 +137,25 @@ RUN \
 
 RUN \
     # install DeepSpeed
-    pip install deepspeed==0.5.7 && \
+    pip install deepspeed==0.6.0 && \
     python -c "import deepspeed; print(deepspeed.__version__)"
 
+RUN \
+    # install Bagua
+    CUDA_VERSION_MM=$(python -c "print(''.join('$CUDA_VERSION'.split('.')[:2]))") && \
```
Comment (carmocca) [inline dockers/base-cuda/Dockerfile:145]: Same comment

File: .azure-pipelines/gpu-tests.yml (hunk 2 above)
Comment (Borda) [reply .azure-pipelines/gpu-tests.yml:57]: how about if version `11.3.1`?

File: dockers/base-conda/Dockerfile (hunk 4 above)
Comment (Borda) [reply dockers/base-conda/Dockerfile:164]: I did and it with this docker and it the message they printed in terminal

File: dockers/base-conda/Dockerfile (hunk 5 above)
Comment (Borda) [reply dockers/base-conda/Dockerfile:152]: we can, but this shall be more stable, related to freezing env

File: .azure-pipelines/gpu-tests.yml (hunk 2 above)
Comment (akihironitta) [reply .azure-pipelines/gpu-tests.yml:57]: By the way, what does `MM` in `CUDA_VERSION_MM` stand for? :eyes:

**Review by carmocca — APPROVED** [review]
